### PR TITLE
Typos flagged by the IBM Quantum Platform automatic spellchecker

### PR DESCRIPTION
**Pull request by abbycross**
### Summary

As part of its build, IQP now has an auto-spellchecker, and it flagged several minor typos in the API refs that are surfaced in IQP from the Qiskit/qiskit repo.  Fixing them in this PR; hope this is the best way to contribute this kind of fix!--feedback welcome! 

### Details and comments

🍎 "Low-hanging fruit" 🍎

### CODE REVIEW

File: qiskit/utils/__init__.py [hunk 1]
```diff
@@ -53,7 +53,7 @@
 backend etc.
 
 
-Optional Depedency Checkers (:mod:`qiskit.utils.optionals`)
+Optional Dependency Checkers (:mod:`qiskit.utils.optionals`)
 ===========================================================
```
Comment (jakelishman) [inline qiskit/utils/__init__.py]:
```suggestion
Optional Dependency Checkers (:mod:`qiskit.utils.optionals`)
============================================================
```

**Review by jakelishman — COMMENTED** [review]
Thanks - these are definitely good to get fixed.

Most of what I'm getting from this is that we _really_ don't know how to spell "overridden" haha.  (I even typed it wrong in that sentence at first, as well...)

File: qiskit/synthesis/clifford/clifford_decompose_greedy.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def synth_clifford_greedy(clifford):
     greedy Clifford compiler that is described in Appendix A of
     Bravyi, Hu, Maslov and Shaydulin.
 
-    This method typically yields better CX cost compared to the Aaronson-Gottesma method.
+    This method typically yields better CX cost compared to the Aaronson-Gottesman method.
```
Comment (Eric-Arellano) [inline qiskit/synthesis/clifford/clifford_decompose_greedy.py:39]: Other reviewers, search results suggest it indeed should be Gottesman but FYI this one is not as obvious.

**Review by Eric-Arellano — APPROVED** [review]
Thank you! This is very helpful. And we appreciate it being one PR rather than 55 small PRs hahaha.

I will mark this as needs-cherrypick so that it gets backported to the 0.44 documentation.

File: qiskit/synthesis/clifford/clifford_decompose_greedy.py (hunk 2 above)
Comment (abbycross) [reply qiskit/synthesis/clifford/clifford_decompose_greedy.py:39]: I lucked out that I have heard this name bandied about in quantum contexts so I had a pretty high degree of confidence  on it 🤣thanks for looking so closely @Eric-Arellano !

**Review by jakelishman — APPROVED** [review]

**Pull request merged**